fleet_adapter_template/fleet_adapter_template/fleet_adapter.py

```diff
@@ -160,44 +162,108 @@ def _updater_inserter(cmd_handle, update_handle):
         """Insert a RobotUpdateHandle."""
         cmd_handle.update_handle = update_handle
 
-    # Initialize robots for this fleet
-    robots = {}
-    for robot_name, robot_config in config_yaml['robots'].items():
-        node.get_logger().info(f"Initializing robot:{robot_name}")
-        rmf_config = robot_config['rmf_config']
-        robot_config = robot_config['robot_config']
-        initial_waypoint = rmf_config['start']['waypoint']
-        initial_orientation = rmf_config['start']['orientation']
-        robot = RobotCommandHandle(
-            name=robot_name,
-            config=robot_config,
-            node=node,
-            graph=nav_graph,
-            vehicle_traits=vehicle_traits,
-            transforms=transforms,
-            map_name=rmf_config['start']['map_name'],
-            initial_waypoint=initial_waypoint,
-            initial_orientation=initial_orientation,
-            charger_waypoint=rmf_config['charger']['waypoint'],
-            update_frequency=rmf_config.get('robot_state_update_frequency', 1),
-            adapter=adapter)
-
-        if robot.initialized:
-            robots[robot_name] = robot
-            # Add robot to fleet
-            fleet_handle.add_robot(robot,
-                                   robot_name,
-                                   profile,
-                                   robot.starts,
-                                   partial(_updater_inserter, robot))
-            node.get_logger().info(
-                f"Successfully added new robot:{robot_name}")
+    # Initialize robot API for this fleet
+    api = RobotAPI(
+        fleet_config['fleet_manager']['prefix'],
+        fleet_config['fleet_manager']['user'],
+        fleet_config['fleet_manager']['password'])
 
-        else:
-            node.get_logger().error(
-                f"Failed to initialize robot:{robot_name}")
+    # Initialize robots for this fleet
 
-    return adapter, fleet_handle, robots
+    missing_robots = config_yaml['robots']
+
+    def _add_fleet_robots():
+        robots = {}
+        while len(missing_robots) > 0:
+            time.sleep(0.2)
+            for robot_name in list(missing_robots.keys()):
```

Comment:
Nit. we do not need to explicitly convert to a list
```suggestion
            for robot_name in missing_robots.keys():
```

Reply:
The original is correct. It' the pythonic way to modifying a dict while iterating over it.